snuba_sdk/expressions.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List, Union
+
+from snuba_sdk import Expression
+
+ScalarType = Union[None, bool, str, float, int, date, datetime]
+
+
+class InvalidExpression(Exception):
+    pass
+
+
+@dataclass(frozen=True)
+class Limit(Expression):
+    limit: int
+
+    def validate(self) -> None:
+        if self.limit < 0 or self.limit > 10000:
+            raise InvalidExpression(f"limit {self.limit} is invalid")
```

Comment:
why not putting these type of structural validation in the _post_init method? A limit object with limit lower than 0 should never be instantiated.

Reply:
The __post_init__ calls `self.validate` so that should already be true.